diff --git a/node_modules/expo-task-manager/android/src/main/java/expo/modules/taskManager/TaskManagerUtils.java b/node_modules/expo-task-manager/android/src/main/java/expo/modules/taskManager/TaskManagerUtils.java
index 4dd0e94..5cadd3f 100644
--- a/node_modules/expo-task-manager/android/src/main/java/expo/modules/taskManager/TaskManagerUtils.java
+++ b/node_modules/expo-task-manager/android/src/main/java/expo/modules/taskManager/TaskManagerUtils.java
@@ -211,7 +211,8 @@ public class TaskManagerUtils implements TaskManagerUtilsInterface {
           .setOverrideDeadline(DEFAULT_OVERRIDE_DEADLINE);
     } else if (Build.VERSION.SDK_INT < 31) {
       // For Android 9 (API 28) to Android 11 (API 30)
-      jobBuilder.setImportantWhileForeground(true);
+      jobBuilder.setImportantWhileForeground(true)
+          .setOverrideDeadline(DEFAULT_OVERRIDE_DEADLINE);
     } else {
       // For Android 12 (API 31) and above
       jobBuilder.setExpedited(true);
